cms/djangoapps/contentstore/views/item.py

```diff
@@ -613,6 +628,14 @@ def _save_xblock(user, xblock, data=None, children_strings=None, metadata=None,
             # IMPORTANT NOTE: if the client passed 'null' (None) for a piece of metadata that means 'remove it'. If
             # the intent is to make it None, use the nullout field
             if metadata is not None:
+                if xblock.category == "course":
+                    discussion_toggle = metadata.pop('discussion_enabled', None)
+                else:
+                    discussion_toggle = metadata.pop('is_discussable', None)
+
+                if discussion_toggle is not None:
+                    set_discussion_toggle(discussion_toggle, xblock, user)
+
```

Comment:
I think you're still conflating these two fields, as I've mentioned in the PR before, setting discussion enabled for a course *should not* alter any other blocks. I think the existing mechanism should handle things, so you can just remove this bit in favour of:

```suggestion
                discussion_toggle = metadata.pop('is_discussable', None)
                if discussion_toggle is not None:
                    set_discussion_toggle(discussion_toggle, xblock, user)

```